test/e2e/specs/editor/blocks/heading.spec.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Heading', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'can be created by prefixing number sign and a space', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '### 3' );
+
+		await expect.poll( editor.getEditedPostContent ).toBe(
```

Comment:
Let's try using `editor.getBlocks` now that we are [recommending](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/e2e/overusing-snapshots.md#the-solution) it over snapshots.

Reply:
Updated most assertions to use the new recommended method but left inline snapshots for color tests.

## Why?

I noticed that `toMatchObject` isn't strict. For example, if I replace the current inline snapshot test with the example below test still passes without checking the injected styles attribute.

```js
await expect.poll( editor.getBlocks ).toMatchObject( [
	{
		name: 'core/heading',
		attributes: { content: 'Heading', level: 3 },
	},
] );
```

Reply:
Yep, because it essentially just does `expect.objectContaining`. I think it's fine though, we could just add the `style` attribute there, since it's something we want to test explicitly.

Reply:
Convert all assertions to use recommended method 👍

Reply:
Thank you! 💯